foundations/src/telemetry/tokio/mod.rs

```diff
@@ -0,0 +1,364 @@
+            self.runtime_id,
+            worker_idx,
+        )
+        .inner()
+        .store(
+            metrics.worker_steal_operations(worker_idx),
+            Ordering::SeqCst,
+        );
+
+        tokio_runtime_worker::polls_total(&self.runtime_name, self.runtime_id, worker_idx)
+            .inner()
+            .store(metrics.worker_poll_count(worker_idx), Ordering::SeqCst);
+
+        tokio_runtime_worker::busy_duration_micros_total(
+            &self.runtime_name,
+            self.runtime_id,
+            worker_idx,
+        )
+        .inner()
+        .store(
+            metrics.worker_total_busy_duration(worker_idx).as_micros() as u64,
+            Ordering::SeqCst,
+        );
+
+        tokio_runtime_worker::local_schedules_total(
+            &self.runtime_name,
+            self.runtime_id,
+            worker_idx,
+        )
+        .inner()
+        .store(
+            metrics.worker_local_schedule_count(worker_idx),
+            Ordering::SeqCst,
+        );
+
+        tokio_runtime_worker::overflows_total(&self.runtime_name, self.runtime_id, worker_idx)
+            .inner()
+            .store(metrics.worker_overflow_count(worker_idx), Ordering::SeqCst);
+
+        tokio_runtime_worker::local_queue_depth(&self.runtime_name, self.runtime_id, worker_idx)
+            .set(metrics.worker_local_queue_depth(worker_idx) as u64);
+
+        tokio_runtime_worker::mean_poll_time_micros(
+            &self.runtime_name,
+            self.runtime_id,
+            worker_idx,
+        )
+        .set(metrics.worker_mean_poll_time(worker_idx).as_micros() as u64);
+    }
+}
+
+#[metrics(crate_path = "crate")]
+mod tokio_runtime {
+    /// Number of worker threads in use by the runtime.
+    ///
+    /// This number shouldn't change during execution.
+    pub(super) fn workers(runtime_name: &Option<Arc<str>>, runtime_id: Option<usize>) -> Gauge;
+
+    /// Current number of blocking threads allocated by the runtime.
+    pub(super) fn blocking_threads(
+        runtime_name: &Option<Arc<str>>,
+        runtime_id: Option<usize>,
+    ) -> Gauge;
+
+    /// Current number of active tasks on the runtime.
+    pub(super) fn active_tasks(runtime_name: &Option<Arc<str>>, runtime_id: Option<usize>)
+        -> Gauge;
+
+    /// Current number of idle blocking threads on the runtime which aren't doing anything.
+    pub(super) fn idle_blocking_threads(
+        runtime_name: &Option<Arc<str>>,
+        runtime_id: Option<usize>,
+    ) -> Gauge;
+
+    /// Counter of schedules not originating from a worker on the runtime.
+    pub(super) fn remote_schedules_total(
+        runtime_name: &Option<Arc<str>>,
+        runtime_id: Option<usize>,
+    ) -> Counter;
+
+    /// Counter of forced yields due to task budgeting.
+    pub(super) fn budget_forced_yields_total(
+        runtime_name: &Option<Arc<str>>,
+        runtime_id: Option<usize>,
+    ) -> Counter;
+
+    pub(super) fn io_driver_fd_registrations_total(
```

Comment:
All the metrics need docs

Reply:
Thanks, this was an oversight, I meant to have docs for these

Reply:
done